databricks/koalas/frame.py

```diff
@@ -361,23 +361,63 @@ def _reduce_for_stat_function(self, sfun, name, axis=None, numeric_only=False):
             return row  # Return first row as a Series
 
         elif axis in ('columns', 1):
-            # Here we execute with the first 1000 to get the return type.
-            # If the records were less than 1000, it uses pandas API directly for a shortcut.
-            limit = 1000
-            pdf = self.head(limit + 1).to_pandas()
-            pser = getattr(pdf, name)(axis=axis, numeric_only=numeric_only)
-            if len(pdf) <= limit:
-                return Series(pser)
+            return self._compute_columns_axis(name, axis=axis, numeric_only=numeric_only)
 
-            @pandas_udf(returnType=as_spark_type(pser.dtype.type))
-            def calculate_columns_axis(*cols):
-                return getattr(pd.concat(cols, axis=1), name)(axis=axis, numeric_only=numeric_only)
+        else:
+            raise ValueError("No axis named %s for object type %s." % (axis, type(axis)))
 
-            df = self._sdf.select(calculate_columns_axis(*self._internal.data_scols).alias("0"))
-            return DataFrame(df)["0"]
+    def _compute_columns_axis(self, name, axis, *args, **kwargs):
```

Comment:
Shall we check the pyspark version and raise an exception if the version is old?